gwcli/ceph.py

```diff
@@ -171,38 +149,99 @@ def ui_command_refresh(self):
         """
 
         self.refresh()
+        self.logger.info("ok")
+
+    def ui_command_info(self):
+
+        color = {"HEALTH_OK": "green",
+                 "HEALTH_WARN": "yellow",
+                 "HEALTH_ERR": "red"}
+
+        status = self.ceph_status
+        osdmap = status['osdmap']['osdmap']
+        output = "Cluster name: {}\n".format(self.cluster_name)
+        output += "Ceph version: {}\n".format(self.version)
+        output += "Health      : {}\n".format(self.health_status)
+        if self.health_status != 'HEALTH_OK':
+            output += " - {}\n".format(','.join(self.health_list))
+
+        mon_names = [mon.get('name') for mon in status["monmap"]["mons"]]
+
+        q_str = "quorum {}".format(','.join(status['quorum_names']))
+        q_out = set(mon_names) - set(status['quorum_names'])
+        if q_out:
+            q_str += ", out of quorum: {}".format(','.join(q_out))
+        output += "\nMONs : {:>4} ({})\n".format(self.topology.num_mons,
+                                                 q_str)
+        output += "OSDs : {:>4} ({} up, {} in)\n".format(self.topology.num_osds,
+                                                        osdmap['num_up_osds'],
+                                                        osdmap['num_in_osds'])
+        output += "Pools: {:>4}\n".format(self.num_pools)
+        raw = status['pgmap']['bytes_total']
+        output += "Raw capacity: {}\n".format(human_size(raw))
+        output += "\nConfig : {}\n".format(self.conf)
+        output += "Keyring: {}\n".format(os.path.join(CephGroup.ceph_config_dir,
+                                                      self.keyring))
+
+        console_message(output, color=color[self.health_status])
+
+    @property
+    def cluster_version(self):
+
+        vers_out = os_cmd("ceph -c {} version".format(self.conf))
+
+        # RHEL packages include additional info, that we don't need
+        version_str = vers_out.split()[2]
+        return '.'.join(version_str.split('.')[:3])
 
     def update_state(self):
+
+        self.logger.debug("Querying ceph for state information")
         with rados.Rados(conffile=self.conf) as cluster:
             cmd = {'prefix': 'status', 'format': 'json'}
             ret, buf_s, out = cluster.mon_command(json.dumps(cmd), b'')
 
-        self.ceph_status = json.loads(buf_s)
-        self.health_status = self.ceph_status['health']['overall_status']
+        status = json.loads(buf_s)
+        self.ceph_status = status
+
+        # assume a luminous or above ceph cluster
+        self.health_status = status['health'].get('status',
+                                                  status['health'].get('overall_status'))
+        self.health_list = [check.get('summary') for check in
```

Comment:
This fails in my test environment w/ v12.1: the contents of ```check.get('summary')``` is ```{u'message': u'application not enabled on 1 pool(s)'}```, which results in a ```TypeError: 'NoneType' object is not iterable``` exception

Reply:
I disabled rbd from my pool, and see the same message from ceph but don't get the exception.

Is mon_health_preluminous_compat=true set ?